python/ray/dag/tests/test_accelerated_dag.py

```diff
@@ -52,6 +57,10 @@ def test_basic(ray_start_regular):
         assert result == i + 1
         output_channel.end_read()
 
+    # Note: must teardown before starting a new Ray session, otherwise you'll get
+    # a segfault from the dangling monitor thread upon the new Ray init.
+    compiled_dag.teardown()
```

Comment:
hmm this seems pretty bad for usability? Should we just call teardown on dag destrutor?

Reply:
It only applies for unit tests, so it's probably ok